src/parser/core/Entity.js

```diff
@@ -38,4 +38,27 @@ export default class Entity {
 			(buff.end === null || (buff.end + bufferTime) >= currentTimestamp) &&
 			(sourceID === null || sourceID === buff.sourceID))
 	}
+
+	/**
+	 * @param {number} forTimestamp Timestamp (in ms) to be considered. Won't work right for timestamps after the currentTimestamp.
+	 * @param {number} bufferTime Time (in ms) after buff's expiration where it will still be included. There's a bug in the combat log where if a spell consumes a buff that buff may disappear a short time before the heal or damage event it's buffing is logged. This can sometimes go up to hundreds of milliseconds.
+	 * @param {number} minimalActiveTime - Time (in ms) the buff must have been active before timestamp for it to be included.
+	 * @param {number} sourceID - source ID the buff must have come from, or any source if null.
+	 * @returns {Object} - An array of buff objects on the target at the given timestamp. The buff object will have all the properties of the associated applybuff event, along with a start timestamp, an end timestamp if the buff has fallen, and an isDebuff flag. If multiple buffs meet the specifications, there's no guarantee which you'll get (this could happen if multiple spells with the same statusId but from different sources are on the same target)
+	 */
+	getStatuses(forTimestamp, bufferTime = 0, minimalActiveTime = 0, sourceID = null) {
+
+
+		const currentTimestamp = forTimestamp < this.parser.currentTimestamp ? forTimestamp
+			: this.parser.currentTimestamp
+
+		if (forTimestamp < this.parser.currentTimestamp) {
+			console.warn('Entity:getStatuses timestamp given is later than current timestamp')
+		}
+
+		return this.buffs.filter(buff =>
```

Comment:
Might be worth consolidating this last bit of logic w/ the `getStatus` method in some way?